src/odemis/driver/orsay.py

```diff
@@ -3141,8 +3161,72 @@ def _setProbeCurrent(self, current: Optional[float]) -> float:
         preset_name = self._probe_current_presets[current]
         # Typically takes < 2s
         self.parent._load_preset(preset_name, PRESET_MASK_NAME, timeout=20)
+
+        # Update the base shift, in case it has changed (in the preset)
+        self._metadata[model.MD_SCAN_OFFSET_CALIB][current] = self._get_base_shift(preset_name)
+
+        # Set back the shift to the current shift (need to pass the current as
+        # .probeCurrent hasn't been updated yet)
+        self._set_shift(self.shift.value, current)
+
         return current
 
+    def _set_shift(self, shift: Tuple[float, float], current: Optional[float] = None) -> Tuple[float, float]:
+        """
+        Setter for .shift
+        :param shift: requested shift (in m)
+        :param current: the probe current for which to set the shift. If None, it will
+        use the .probeCurrent value.
+        :return: actual shift set (in m)
+        """
+        if current is None:
+            current = self.probeCurrent.value
+        try:
+            base_shift = self._metadata[model.MD_SCAN_OFFSET_CALIB][current]
+        except KeyError as ex:
+            logging.debug("No base shift known for current %s, using (0,0)", ex)
+            base_shift = (0, 0)
+
+        # Add the base shift, and make sure it still fits the range
+        act_shift = base_shift[0] + shift[0], base_shift[1] + shift[1]
+        act_shift = self._fib_beam.shift.clip(act_shift)
+        self._fib_beam.shift.value = act_shift
+
+        shift_clipped = act_shift[0] - base_shift[0], act_shift[1] - base_shift[1]
+        if shift != shift_clipped:
+            logging.info("Requested shift %s clipped to %s", shift, shift_clipped)
+        return shift_clipped
```

Comment:
Will not the clipped shift have an an impact on drift correction? In such a case where we have clipped shift, when it compensates for drift, it will not able to compensate because it is at the boundary limit. In that case, we maybe can stop the milling and ask the user to resume milling after the shift has been reset by Odemis.

Reply:
Indeed, that's quite true in general with drift correction, if we reach the limit of the shift we need to "gracefully" handle it. For now, on the SPARC, we generate a warning, and continue hoping for the best (which usually is not happening). For milling that could be even worse (because you can re-acquire after a failed acquisition, but you cannot undo a blurring hole). So we probably should detect that the shift reached its limit, and stop the milling of the given pattern (an alternative being to compensate with stage move, but that's for the future). 

I'd suggest to do the following two things in the milling:
* at the beginning of a new feature (aka site), reset the shift to 0,0
* Every time the shift is applied, check it's within bounds, and accepted. If not, log an error, and move to the next site.

Reply:
How do we track this issue? For e.g. through a TODO or a task.